core/crud/crud.service.ts

```diff
@@ -295,34 +296,45 @@ export class CrudService<T extends CrudEntity> {
       } as CrudContext;
     }
 
-    const res = await axios
-      .patch(url, payload, {
-        params: query,
-        auth: {
-          username: this.crudConfig.microServicesOptions.username,
-          password: this.crudConfig.microServicesOptions.password,
-        },
-      })
-      .catch((e) => {
-        const error = e.response?.data || e;
-        throw new HttpException(
-          {
-            statusCode: error.statusCode,
-            error: error.error,
-            message: error.message,
+    const retries: number = 2;
+    let attempt: number = 0;
+    const delayStartValue: number = 100;
+    while (attempt < retries) {
+      try {
+        const res = await axios.patch(url, payload, {
+          params: query,
+          auth: {
+            username: this.crudConfig.microServicesOptions.username,
+            password: this.crudConfig.microServicesOptions.password,
           },
-          error.statusCode,
-        );
-      });
+        });
+
+        const result = res.data.res;
+        const partialCtx = res.data.ctx;
+        if (partialCtx && ctxPos != null && args[ctxPos]) {
+          for (const key in partialCtx) {
+            args[ctxPos][key] = partialCtx[key];
+          }
+        }
+        return result;
+      } catch (e) {
+        ++attempt;
+        const error = e.response?.data || e;
 
-    const result = res.data.res;
-    const partialCtx = res.data.ctx;
-    if (partialCtx && ctxPos != null && args[ctxPos]) {
-      for (const key in partialCtx) {
-        args[ctxPos][key] = partialCtx[key];
+        if (!isNetworkError(e) || attempt > retries) {
+          throw new HttpException(
+            {
+              statusCode: error.statusCode,
+              error: error.error,
+              message: error.message,
+            },
+            error.statusCode,
+          );
```

Comment:
You can do `throw e` here for better visibility.

Additionally, to avoid recursive retries, a non-network error should be thrown in case of `attempt > retries`. A bad request or a special code could do.

To clarify: we don't want `ms-A` to retry 3 times, then throw a network error, for `ms-B` to retry calling `ms-A` 3 times (and so on).

Reply:
Also it seems that the package you are using only check for browser errors. Since we're in a node.js environment it will not work.

My best bet would be to check the Axios docs: https://github.com/axios/axios/issues/2986